src/platform.ts

```diff
@@ -55,196 +60,437 @@ export function getPlatformDetails(): IPlatformDetails {
 
-const powerShell64BitPathOn32Bit = SysnativePowerShellPath.toLocaleLowerCase();
-const powerShell32BitPathOn64Bit = SysWow64PowerShellPath.toLocaleLowerCase();
+    private findPSCoreDotnetGlobalTool(): IPossiblePowerShellExe {
+        const exeName: string = this.platformDetails.operatingSystem === OperatingSystem.Windows
+            ? "pwsh.exe"
+            : "pwsh";
 
-export function fixWindowsPowerShellPath(powerShellExePath: string, platformDetails: IPlatformDetails): string {
-    const lowerCasedPath = powerShellExePath.toLocaleLowerCase();
+        const dotnetGlobalToolExePath: string = path.join(os.homedir(), ".dotnet", "tools", exeName);
 
-    if ((platformDetails.isProcess64Bit && (lowerCasedPath === powerShell64BitPathOn32Bit)) ||
-        (!platformDetails.isProcess64Bit && (lowerCasedPath === powerShell32BitPathOn64Bit))) {
-            return System32PowerShellPath;
+        return new PossiblePowerShellExe(dotnetGlobalToolExePath, ".NET Core PowerShell Global Tool");
     }
 
-    // If the path doesn't need to be fixed, return the original
-    return powerShellExePath;
-}
+    private findPSCoreMsix(options?: { findPreview: boolean }): IPossiblePowerShellExe {
+        // We can't proceed if there's no LOCALAPPDATA path
+        if (!process.env.LOCALAPPDATA) {
+            return null;
+        }
 
-/**
- * Gets a list of all available PowerShell instance on the specified platform.
- * @param platformDetails Specifies information about the platform - primarily the operating system.
- * @param sessionSettings Specifies the user/workspace settings. Additional PowerShell exe paths loaded from settings.
- * @returns An array of IPowerShellExeDetails objects with the PowerShell name & exe path for each instance found.
- */
-export function getAvailablePowerShellExes(
-    platformDetails: IPlatformDetails,
-    sessionSettings: Settings.ISettings | undefined): IPowerShellExeDetails[] {
-
-    let paths: IPowerShellExeDetails[] = [];
-
-    if (platformDetails.operatingSystem === OperatingSystem.Windows) {
-        if (platformDetails.isProcess64Bit) {
-            paths.push({
-                versionName: WindowsPowerShell64BitLabel,
-                exePath: System32PowerShellPath,
-            });
-
-            paths.push({
-                versionName: WindowsPowerShell32BitLabel,
-                exePath: SysWow64PowerShellPath,
-            });
+        // Find the base directory for MSIX application exe shortcuts
+        const msixAppDir = path.join(process.env.LOCALAPPDATA, "Microsoft", "WindowsApps");
+
+        if (!fs.existsSync(msixAppDir)) {
+            return null;
+        }
+
+        // Define whether we're looking for the preview or the stable
+        let pwshMsixDirRegex: RegExp = null;
+        let pwshMsixName: string = null;
+        if (options && options.findPreview) {
+            pwshMsixDirRegex = PowerShellExeFinder.PwshPreviewMsixRegex;
+            pwshMsixName = "PowerShell Preview MSIX";
```

Comment:
If the user installs it from the store, they don't need to know what an "MSIX" is. Maybe from a users perspective we call it "App" or "Windows App"?

cc @SydneyhSmith

Reply:
Yeah I'd like to know what the right name here is. Discussed with @SteveL-MSFT initially and defaulted to `MSIX` but would like to improve the name

Reply:
@joeyaiello what would you call this?